src/Components/ArtistFilter/Artworks.tsx

```diff
@@ -14,41 +9,91 @@ import { Artworks_filtered_artworks } from "../../__generated__/Artworks_filtere
 
 interface Props {
   filtered_artworks: Artworks_filtered_artworks
-  relay: RelayPaginationProp
+  relay: RelayRefetchProp
   artistID: string
 }
 
 const PAGE_SIZE = 10
 
 class Artworks extends React.Component<Props> {
-  loadMoreArtworks(filters) {
+  loadPage(filters, page) {
+    this.props.relay.refetch(
+      {
+        page,
+        size: PAGE_SIZE,
+        filteredArtworksNodeID: this.props.filtered_artworks.__id,
+      },
+      null,
+      error => {
+        if (error) {
+          console.error(error)
+        }
+        filters.setPage(page)
+      }
+    )
+  }
+
+  loadNextPage(filters) {
     const hasMore = this.props.filtered_artworks.artworks.pageInfo.hasNextPage
-    const endCursor = this.props.filtered_artworks.artworks.pageInfo.endCursor
+
     if (hasMore) {
-      // TODO: Should refetchConnection keep appending records?
-      this.props.relay.refetchConnection(
-        PAGE_SIZE,
-        error => {
-          if (error) {
-            console.error(error)
-          }
-          filters.incrementPage()
-        },
-        { cursor: endCursor }
+      return this.loadPage(filters, filters.state.page + 1)
+    }
+  }
+
+  loadPrevPage(filters) {
+    if (filters.state.page > 1) {
+      return this.loadPage(filters, filters.state.page - 1)
+    }
+  }
+
+  renderPaginationBar(filters) {
```

Comment:
This just renders all the numbers from 1 up until `last page`, and lets you click to select any particular one. Basically just to demonstrate that functionality, the real pagination bar component will have a bit of logic as to which pages to display (1... 6, 7, 8 .... 19) kinda thing.